app/packages/state/src/hooks/useHover.ts

```diff
@@ -0,0 +1,26 @@
+import { useRef, useState, useEffect } from "react";
+
+function useHover() {
+  const [value, setValue] = useState(false);
+  const ref = useRef(null);
+  const handleMouseOver = () => setValue(true);
+  const handleMouseOut = () => setValue(false);
+
+  useEffect(
+    () => {
+      const node = ref.current;
+      if (node) {
+        node.addEventListener("mouseover", handleMouseOver);
+        node.addEventListener("mouseout", handleMouseOut);
+        return () => {
+          node.removeEventListener("mouseover", handleMouseOver);
+          node.removeEventListener("mouseout", handleMouseOut);
+        };
+      }
+    },
+    [ref.current] // Recall only if ref changes
```

Comment:
It might be safer to have this hook return `onMouseOver` and`onMouseOut` event handlers rather the ref. Just because the ref itself won't trigger updates